File: acceptance/Sagar#9818/13-roman_to_integer.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int romanToInt(string s) {

        int n = s.length()-1;
        int ans = 0;

        auto value = [](char ch)->int{
            switch(ch)
            {
                case 'M': return 1000;
                case 'D': return 500;
                case 'C': return 100;
                case 'L': return 50;
                case 'X': return 10;
                case 'V': return 5; 
                case 'I': return 1;
            }

            return 0;
        };

        int temp = 0;

        for( int i=n; i>=0; --i )
        {
            auto val = value(s[i]);
            if( val>=temp )
                ans += val;
            else 
                ans -= val;

            temp = val;
        }

        return ans;
    }
};

File: acceptance/csangam/13 - roman to integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanValue={
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000
        }

        revRoman=s[::-1]

        value=0 + romanValue[revRoman[0]]

        for i in range(1,len(revRoman)):
            if romanValue[revRoman[i]]<romanValue[revRoman[i-1]]:
                value-=romanValue[revRoman[i]]
            else:
                value+=romanValue[revRoman[i]]

        return value
